### DISCUSSION TITLE
You can improve the mixin _forms.scss

### DISCUSSION THREAD

**Discussion by maikycourse** [twbs/bootstrap#30850] [General]
line: 33
`.was-validated #{if(&, "&", "")}[required]:#{$state},`
I have added "[required]" to prevent it from affecting inputs that are not required. 

I think the code that is generated by adding "[required]" is more appropriate.
I would like you to take it into account in future versions.
Thanks.

**Comment by mdo** [comment]
[From the forms docs page:](https://getbootstrap.com/docs/4.5/components/forms/#validation)

> Bootstrap scopes the `:invalid` and `:valid` styles to parent `.was-validated` class, usually applied to the `<form>`.  Otherwise, any required field without a value shows up as invalid on page load. This way, you may choose when to activate them (typically after form submission is attempted).

**Reply by maikycourse** [reply]
Sin modificar el mixin, dejando el código original.
Ejemplo:
**Formulario**
- email (required)
- teléfono (no required)
- comentario (required)
**Paso 1**
- No relleno el formulario y le doy a enviar
**Paso 2**
- El formulario muestra:
   - email (borde rojo)
   - teléfono (borde verde)
   - comentario (borde rojo)
**Conclusión**
No quiero que teléfono se muestre con borde verde, ya que dejarlo vacio no está bien, pero como no es obligatorio debería de mostrarse con el aspecto por defecto de un input.

**Traducción Google translate**
> Without modifying the mixin, leaving the original code.
> Example:
> **Form**
> - email (required)
> - phone (not required)
> - comment (required)
> **Step 1**
> - I do not fill out the form and I give it to send
> **Step 2**
> - The form shows:
>     - email (red border)
>     - phone (green border)
>     - comment (red border)
> **Conclusion**
> I do not want the phone to be shown with a green border, because leaving it empty is not good, but since it is not mandatory it should be shown with the default aspect of an input.

**Comment by maikycourse** [comment]
https://codepen.io/maikyabelico/pen/BaovgzM
Look at my example, to understand better.
Thanks
